lib/plugins/filter/post_permalink.js

```diff
@@ -6,7 +6,13 @@ let permalink;
 
 function postPermalinkFilter(data) {
   const { config } = this;
-  const { id, _id, slug, title, date } = data;
+  const { id, _id, slug, title, date, __permalink } = data;
+
+  if (__permalink) {
+    if (!__permalink.startsWith('/')) return `/${__permalink}`;
+    return __permalink;
+  }
```

Comment:
When a `__permalink` is detected (which is added by post processor from `permalink` inside the front-matter), it will be return directly.